File: src/main/java/aristata/programmers/d20190316/Camouflage.java
package aristata.programmers.d20190316;

import java.util.Arrays;
import java.util.HashMap;

public class Camouflage {

    private int solution(String[][] clothes) {
        int answer = 1;

        HashMap<String, Integer> map = new HashMap<>();
        Arrays.stream(clothes).forEach(it ->
                map.put(it[1], map.getOrDefault(it[1], 0) + 1)
        );

        // key - 개수
        // 공식 => 1 * (개수 + 1) [* (개수 + 1)...n] -1
        for (int a : map.values()) {
            answer = answer * (a + 1);
        }

        return answer - 1;
    }

    public static void main(String[] args) {

        String[][] testCase1 = {
                {"yellow_hat", "headgear"},
                {"blue_sunglasses", "eyewear"},
                {"green_turban", "headgear"}
        };

        String[][] testCase2 = {
                {"crow_mask", "face"},
                {"blue_sunglasses", "face"},
                {"smoky_makeup", "face"}
        };

        Camouflage camouflage = new Camouflage();
        int result = camouflage.solution(testCase1);
        System.out.println("결과: " + result);

        int result2 = camouflage.solution(testCase2);
        System.out.println("결과2: " + result2);
    }


}

File: src/main/kotlin/aristata/programmers/d20190316/Camouflage.kt
package aristata.programmers.d20190316

class Camouflage {

    fun solution(clothes: Array<Array<String>>): Int {
        var answer = 1

        val clothMap = mutableMapOf<String, Int>()
        clothes.forEach {
            clothMap[it[1]] = clothMap.getOrDefault(it[1], 0) + 1
        }
        for (count in clothMap.values) {
            answer *= (count + 1)
        }

        return answer - 1
    }
}

fun main(args: Array<String>) {

    val testCase1 = arrayOf(
            arrayOf("yellow_hat", "headgear"),
            arrayOf("blue_sunglasses", "eyewear"),
            arrayOf("green_turban", "headgear")
    )

    val testCase2 = arrayOf(
            arrayOf("crow_mask", "face"),
            arrayOf("blue_sunglasses", "face"),
            arrayOf("smoky_makeup", "face")
    )

    val camouflage = Camouflage()
    val result = camouflage.solution(testCase1)
    println("결과: $result")

    val result2 = camouflage.solution(testCase2)
    println("결과2: $result2")
}

